### ISSUE TITLE
Zoom about a node

### ISSUE THREAD

**Issue by aritraghosh** [jacomyal/sigma.js#833]
Is there a way to zoom about a point? 

I have seen [Issue 227](https://github.com/jacomyal/sigma.js/issues/227) and it zooms about the centre of the canvas. Is it possible to zoom about a custom point meaning anchor at a point and zoom the rest

**Comment by Yomguithereal** [comment]
Hello @aritraghosh. Check out [this](https://github.com/medialab/toflit18/blob/master/client/js/components/exploration/viz/Network.jsx#L40-L52) which focus the camera on a particular node. You'll have to find your camera & its name however (I think its `0` by default).

**Comment by aritraghosh** [comment]
@Yomguithereal That was helpful. While this does zoom about the node it brings it into the centre of the canvas. Is there some way to get the node back to where it was before zooming?

**Comment by Yomguithereal** [comment]
You are moving the camera here, not the node so I am not sure I understand you question. If you mean how you get back to previous camera state then just store it in a variable and reapply it later when needed.

**Comment by aritraghosh** [comment]
Apologies for the confusion. What I want in essence is to keep the left bottom most node fixed (anchored) and zoom the rest of the graph about it.  Is  there way to do this

**Comment by Yomguithereal** [comment]
I am not sure to visualize what you mean.

**Comment by aritraghosh** [comment]
When the zoom happens , it happens about a point. This in general happens to the centre of the canvas. When we do what you suggested, it brings the node to the centre of the canvas and then zooms about it. Is there some way to get the point to where it was before zooming.

**Comment by Yomguithereal** [comment]
Ok. The confusion comes from the fact you speak about moving the point whereas what you are moving is really the camera. So, to do what you need, just store the previous state of the camera and reapply it to return to the original position.

**Comment by aritraghosh** [comment]
Thanks. I am facing a related issue. Is it possible to translate a graph i.e moving a graph using the JS

**Comment by Yomguithereal** [comment]
Yes. Do you want to move the graph (nodes' positions) or the camera (way you look at the graph)?

**Comment by aritraghosh** [comment]
Either is ok . From what I understand  moving the graph is possible by specifying the x and y and ratio. However I would like to move the camera based on a nodes position

**Comment by Yomguithereal** [comment]
Then you need to find the x and y position according to the camera of your node (see the first link I showed you). Then apply this x and y to the camera and you're done.

**Comment by aritraghosh** [comment]
Is there a way to move the graph just in case?

**Comment by Yomguithereal** [comment]
Yeah, sure. Just iterate over the node, change their x or y position and refresh the graph.

```js
s.graph.nodes().forEach(node => {
  node.x += 10;
});

s.refresh();
```

But you'll probably need to disable auto rescaling of the graph in the settings.
